tests/p2p/test_p2p.py

```diff
@@ -182,50 +216,125 @@ def test_many_gossip_messages(setup_clients, add_curl):
         out = api_call(client_ip, data, api, testconfig['namespace'])
-        assert "{'value': 'ok'}" in out
+        expected_ret = expected_ret
+        ass_err = f"test_invalid_msg: expected \"{expected_ret}\" and got \"{out}\""
+        assert expected_ret in out, ass_err
 
-    gossip_propagation_sleep = (TEST_MESSAGES + 20) * timeout_factor  # currently we expect short propagation times.
+    # currently we expect short propagation times.
+    gossip_propagation_sleep = (num_of_msg + prop_sleep_time) * timeout_factor
     print('sleep for {0} sec to enable gossip propagation'.format(gossip_propagation_sleep))
     time.sleep(gossip_propagation_sleep)
 
-    total_expected_gossip = initial + len(setup_clients.pods) * TEST_MESSAGES
     after = poll_query_message(indx=current_index,
                                namespace=testconfig['namespace'],
                                client_po_name=setup_clients.deployment_name,
-                               fields=fields,
+                               fields=headers,
                                findFails=False,
                                expected=total_expected_gossip)
 
-    assert total_expected_gossip == len(after), "test_many_gossip_sim: Total gossip messages in ES is not as expected"
+    err_msg = "msg_testing: Total gossip messages in ES is not as expected"
+    err_msg += f"\nexpected {total_expected_gossip}, got {len(after)}"
+    assert total_expected_gossip == len(after), err_msg
+
+
+# Deploy X peers
+# Wait for bootstrap
+# Broadcast Y messages (make sure that all of them are live simultaneously)
+# Validate that all nodes got exactly Y messages (X*Y messages)
+# Sample few nodes and validate that they got all 5 messages
+def test_many_gossip_sim(setup_clients, add_curl):
+    api = 'v1/broadcast'
+    headers = {'M': 'new_gossip_message', 'protocol': 'api_test_gossip'}
+    msg_size = 10000  # 1kb TODO: increase up to 2mb
+    test_messages = 100
+    pods_num = len(setup_clients.pods)
+
+    prev_num_of_msg = len(query_message(current_index, testconfig['namespace'], setup_clients.deployment_name, headers))
+    # if msg is valid we should see the message at each node msg * pods(nodes)
+    total_expected_gossip = prev_num_of_msg + test_messages * pods_num
+
+    send_msgs(setup_clients, api, headers, total_expected_gossip, num_of_msg=test_messages)
+
+
+def test_broadcast_unknown_protocol(setup_bootstrap, setup_clients, add_curl):
+    api = 'v1/broadcast'
+    # protocol is modified
+    headers = {'M': 'new_gossip_message', 'protocol': 'unknown_protocol'}
+    msg_size = 10000  # 1kb TODO: increase up to 2mb
+    test_messages = 10
+
+    prev_num_of_msg = len(query_message(current_index, testconfig['namespace'], setup_clients.deployment_name, headers))
+    # add only the number of previous messages
+    # when there's a problem in our protocol we're not even sending
+    total_expected_gossip = prev_num_of_msg
```

Comment:
is this atomic? it seems like number of previous messages can still change between now and when we call send_msgs

Reply:
this is how Yosher tested it so far, tests don't run parallel to each other and on p2p tests we don't send traffic without sleeping for propagation so I think we're good that way here.

Reply:
currently we use the same network for these tests serially, since they are from the same type it is hard to do it atomically as their logs look exactly the same. if we could predict our hashes (hash on the tests as well, or get hash back from the API) we could do it. the other idea I don't like is to manipulate the node so it can register new protocols on the flight. that way we can differentiate with protocol names.